pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -412,10 +403,12 @@ func (dswp *desiredStateOfWorldPopulator) checkVolumeFSResize(
 			"as the volume is mounted as readonly", podVolume.Name, pod.Namespace, pod.Name)
 		return
 	}
-	if volumeRequiresFSResize(pvc, volumeSpec.PersistentVolume) {
-		dswp.actualStateOfWorld.MarkFSResizeRequired(uniqueVolumeName, uniquePodName)
-	}
-	processedVolumesForFSResize.Insert(string(uniqueVolumeName))
+	pvCap := volumeSpec.PersistentVolume.Spec.Capacity.Storage()
+	pvcStatusCap := pvc.Status.Capacity.Storage()
+	dswp.desiredStateOfWorld.UpdatePersistentVolumeSize(uniqueVolumeName, *pvCap)
+
+	// in case the actualStateOfWorld was rebuild after kubelet restart ensure that claimSize is set to accurate value
+	dswp.actualStateOfWorld.SetVolumeClaimSize(uniqueVolumeName, *pvcStatusCap)
```

Comment:
this seems odd. We update asw in dswp? This is because reconstruction doesn't know/have the pvc status? CSI at least keeps a file, so we could use that?

Reply:
Yes, that is indeed the case. I would like to this more about recording pvc's status size in json file. Can I do this in a follow up?

Reply:
sure can you open an issue to track

Reply:
https://github.com/kubernetes/kubernetes/issues/92947